### PR TITLE
Backport jwkutil changes

### PR DESCRIPTION
**Pull request by moskyb**
We made some changes to the `jwkutil` package in https://github.com/buildkite/agent/pull/2481, this PR pulls those changes in and adds some doc comments

### CODE REVIEW

File: jwkutil/validate.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+)
+
+var (
+	ErrKeyMissingAlg             = errors.New("key is missing algorithm")
+	ErrUnsupportedAlgForKeyType  = errors.New("unsupported key type")
+	ErrInvalidSigningAlgorithm   = errors.New("invalid signing algorithm")
+	ErrUnsupportSigningAlgorithm = errors.New("unsupported signing algorithm for key type")
+)
+
+func Validate(key jwk.Key) error {
```
Comment (DrJosh9000) [inline jwkutil/validate.go:34]: hmm, this one was missed? Validate seems deserving!

**Review by DrJosh9000 — APPROVED** [review]

File: jwkutil/validate.go (hunk 1 above)
Comment (DrJosh9000) [reply jwkutil/validate.go:34]: Oh, missed upstream in agent.

Comment (moskyb) [reply jwkutil/validate.go:34]: i'll get something in there :)

File: signature/steps.go [hunk 2]
```diff
@@ -45,6 +44,7 @@ func SignSteps(s pipeline.Steps, key jwk.Key, env map[string]string, pInv *Pipel
 	return nil
 }
 
+// SignPipeline adds signatures to each command step (and recursively to any command steps that are within group steps) within a pipeline
 func SignPipeline(p *pipeline.Pipeline, key jwk.Key, pInv *PipelineInvariants) error {
```
[the remark this answers is not in the document]
Comment (triarius) [reply signature/steps.go:48]: Never mind, this also bumps into the convention of putting the mutated arguments first 🤷‍♂️

[the remark this answers is not in the document]
Comment (moskyb) [reply signature/steps.go:48]: yeah, i had that decision process when i was writing it - the way i was looking at it, it's as close as i can get to the `Steps` or the `Pipeline` being the receiver of the method

**Pull request merged**